src/System.Text.Json/tests/Serialization/SnakeCaseUnitTests.cs

```diff
@@ -0,0 +1,141 @@
+            Assert.Equal("url", ConvertToSnakeCase("URL"));
+            Assert.Equal("id", ConvertToSnakeCase("ID"));
+            Assert.Equal("i", ConvertToSnakeCase("I"));
+            Assert.Equal("", ConvertToSnakeCase(""));
+            Assert.Null(ConvertToSnakeCase(null));
+            Assert.Equal("person", ConvertToSnakeCase("Person"));
+            Assert.Equal("i_phone", ConvertToSnakeCase("iPhone"));
+            Assert.Equal("i_phone", ConvertToSnakeCase("IPhone"));
+            Assert.Equal("i_phone", ConvertToSnakeCase("I Phone"));
+            Assert.Equal("i_phone", ConvertToSnakeCase("I  Phone"));
+            Assert.Equal("i_phone", ConvertToSnakeCase(" IPhone"));
+            Assert.Equal("i_phone", ConvertToSnakeCase(" IPhone "));
+            Assert.Equal("is_cia", ConvertToSnakeCase("IsCIA"));
+            Assert.Equal("vm_q", ConvertToSnakeCase("VmQ"));
+            Assert.Equal("xml2json", ConvertToSnakeCase("Xml2Json"));
+            Assert.Equal("sn_ak_ec_as_e", ConvertToSnakeCase("SnAkEcAsE"));
+            Assert.Equal("sn_a__k_ec_as_e", ConvertToSnakeCase("SnA__kEcAsE"));
+            Assert.Equal("sn_a__k_ec_as_e", ConvertToSnakeCase("SnA__ kEcAsE"));
+            Assert.Equal("already_snake_case_", ConvertToSnakeCase("already_snake_case_ "));
+            Assert.Equal("is_json_property", ConvertToSnakeCase("IsJSONProperty"));
+            Assert.Equal("shouting_case", ConvertToSnakeCase("SHOUTING_CASE"));
+            Assert.Equal("9999_12_31t23_59_59_9999999z", ConvertToSnakeCase("9999-12-31T23:59:59.9999999Z"));
+            Assert.Equal("hi_this_is_text_time_to_test", ConvertToSnakeCase("Hi!! This is text. Time to test."));
+            Assert.Equal("building", ConvertToSnakeCase("BUILDING"));
+            Assert.Equal("building_property", ConvertToSnakeCase("BUILDING Property"));
+            Assert.Equal("building_property", ConvertToSnakeCase("Building Property"));
+            Assert.Equal("building_property", ConvertToSnakeCase("BUILDING PROPERTY"));
+        }
+
+        // Use a helper method since the method is not public.
+        private static string ConvertToSnakeCase(string name)
+        {
+            JsonNamingPolicy policy = JsonNamingPolicy.SnakeCase;
+            string value = policy.ConvertName(name);
+            return value;
+        }
+
+        [Theory]
+        [MemberData(nameof(NamesToTest))]
+        public static void Convert_ProperyName_MatchesNewtonsoft(string name)
+        {
+            JsonNamingPolicy policy = JsonNamingPolicy.SnakeCase;
+            NamingStrategy strategy = new SnakeCaseNamingStrategy();
+
+            string actual = policy.ConvertName(name);
+            string expected = strategy.GetPropertyName(name, false);
+
+            Assert.Equal(expected, actual);
+        }
+
+        [Theory]
+        [MemberData(nameof(NamesToTest))]
+        public static void Convert_DictionaryKey_MatchesNewtonsoft(string name)
```

Comment:
For  completeness, can we also add tests where the `JsonSerializer` is used to serialize POCOs with this naming policy passed in as `JsonSerializerOptions`?